pkg/views/styles.go

```diff
@@ -61,6 +72,45 @@ func GetStyledSelectList(items []list.Item, parentIdentifier ...string) list.Mod
 
 	l := list.New(items, d, 0, 0)
 
+	if listOptions != nil && !listOptions[0].IsPaginationDisabled {
+		// Sets the mouse cursor to point to the first index of newly loaded items
+		if listOptions[0].CursorIndex > 0 {
+			l.Select(listOptions[0].CursorIndex)
+		}
+
+		// Add the 'Load More' option in search filter results
+		l.Filter = func(term string, targets []string) []list.Rank {
+			ranks := list.DefaultFilter(term, targets)
+
+			loadMoreIdx := -1
+			// Ideally 'Load More' option if present should be found at the last index
+			for i := len(targets) - 1; i >= 0; i-- {
+				if targets[i] == ListNavigationRenderText {
+					loadMoreIdx = i
+					break
+				}
+			}
+
+			if loadMoreIdx == -1 {
+				return ranks
+			}
+
+			// Return if already present
+			for i := range ranks {
+				if ranks[i].Index == loadMoreIdx {
+					return ranks
+				}
+			}
+
+			// Append 'Load More' option in search filter results
+			ranks = append(ranks, list.Rank{
+				Index: loadMoreIdx,
+			})
+
+			return ranks
+		}
+	}
+
```

Comment:
Not sure where to comment this, but I believe somewhere here the following bug happens:
Let's say `perPage` is set to `10` instead of `100` and user has 33 repositories either personal or within organization, doesn't matter.
What happens is, once the user is on `page=3` and fetched `items=30` user is shown a `+ Load more` option (3 items are still to be fetched with it). When user selects the `+ Load more` option, remaining items are fetched but it seems that if the number of fetched items is less than `perPage` value user is thrown back at the beginning (`page=1`, 1st element).
This is a bug as no matter how many elements are fetched (less, equal or more than the `perPage` vlaue), next selected index should be that of the first element of the newly fetched items.

If this needs more clarification, I would be more than happy to reiterate.

Reply:
thanks for finding this out, have fixed this.

Reply:
Can you please run `daytona purge` and test everything once again from scratch?
I'm getting and `Unauthorized` error now after your rebase with main.
Steps to reproduce are as follows:

> 1. Run `daytona purge` (stop server if running before)
> 2. Run `daytona serve`
> 3. Run `daytona gp add`
> 3.1. Select GitHub and authorize it
> 4. Run`daytona create`
> 4.1. Select added git provider and observer error

Please, get back to me then

Reply:
oops, i had typed about finding the same thing, but never pressed sent button. 
Yeah, I found things are not working after rebase

Reply:
Let's tend to things that are not working then.
Please, purge daytona before testing it thoroughly and once you think it is ready for me to test it again, ping me. :D

Reply:
hey, is something broken in the latest main, I have fixed everything except for 1 thing: On submitting request for creation of a new workspace it throws :  `level=fatal msg="failed to create workspace: git provider not found"`,  

Found same behaviour in main branch  too..

Reply:
Have you purged daytona before testing so you can start fresh?
Everything works for me on main and prod. Can you record a video of the bug? Please start the video with purging daytona just to make sure.

Reply:
wow! So when i started recording to post here, it worked, The only difference here was removing all existing gp identities and adding a new one made it worked it seems. (Previously also, I was creating a new identity and using it but never removed existing ones.)

You can review, but above thing can perhaps need a new ticket (say during purge we also remove existing git provider identities ?)

Reply:
I told you twice to purge and you didn't before, that's why it didn't work. :)
Let's follow instructions in future, no need to raise an issue for now.
I tested it twice and everything works. Good job.

Reply:
I always did purged, but removing the existing gp identities was the only diff. Sure about this 😅